src/compiler/moduleNameResolver.ts

```diff
@@ -2092,10 +2092,11 @@ namespace ts {
     }
 
     /**
+     * @internal
      * From https://github.com/nodejs/node/blob/8f39f51cbbd3b2de14b9ee896e26421cc5b20121/lib/internal/modules/esm/resolve.js#L722 -
      * "longest" has some nuance as to what "longest" means in the presence of pattern trailers
      */
-    function comparePatternKeys(a: string, b: string) {
+    export function comparePatternKeys(a: string, b: string) {

```

Comment:
Question for @weswigham: is this observably different from the way we pick the longest match for `paths`/`typesVersions` patterns with wildcards?

Reply:
I think it's different with how it sorts pattern matches with characters after the `*`, yeah. We sort by absolute length of the final match, iirc, but this sorts on prefix length first and then suffix length.